src/pip/_internal/utils/hashes.py

```diff
@@ -82,7 +82,7 @@ def check_against_chunks(self, chunks: Iterable[bytes]) -> None:
                 hash.update(chunk)
 
         for hash_name, got in gots.items():
-            if got.hexdigest() in self._allowed[hash_name]:
+            if got.hexdigest() in [x.lower() for x in self._allowed[hash_name]]:
```

Comment:
I feel we should convert more eagerly when the values go into `_allowed`.

Reply:
I had initially lowercased prior to storing in `_allowed`, but it ends up creating a larger overall change and changes an unrelated test because it loses track of the case of what the user initial supplied, and therefore the error message. If that's preferable, I can go back to that behavior.

Reply:
I update the code to lowercase the hashes prior to adding them to `_allowed`.